app/core/model/src/main/java/io/syndesis/model/log/LogStep.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.model.log;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.syndesis.model.integration.Step;
+import org.immutables.value.Value;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Value.Immutable
+@JsonDeserialize(builder = LogStep.Builder.class)
+@SuppressWarnings("immutables")
+public interface LogStep extends Step {
```

Comment:
So how come we need to add a new model object for log?  End users seem to be able to add steps to syndesis via extensions without changing the model.  Why can we do something similar?  @lburgazzoli  what do you think?

Reply:
I'd avid adding a specific class for such things as we are now in a situation were we have some untyped steps (endpoint) and some typed ones (*Filters) which make the model inconsistent so we should decide to go with all untyped or all typed.

To me this class looks like an helper class to extract things from getConfiguredProperties and the helpers can be added to LogStepHandler.

Reply:
I vote to go as untyped as possible to keep it inline with what extensions are allowed to do.
@iocanel ??